### PR TITLE
[DATA-8793] Fix stack deploy when the job_id in status.json are not found in Databricks workspace

### PR DESCRIPTION
**Pull request by catherineta**
1. Make changes to abort the deployment and print out a warning message when there is discrepancy between job_id's existence in status file and deletion from workspace. Then it is up to the user to either delete the current status files or pulling the latest status file from other version control sources (e.g. github).
2. Add Test Case #5 to make sure this fix works.

### CODE REVIEW

File: tests/stack/test_api.py [hunk 1]
```diff
@@ -140,6 +143,7 @@
                          ]
 }
 
+job_id_deleted = 111
```
Comment (potcharapol-databricks) [inline tests/stack/test_api.py]: Maybe put as a class state (like `self.job_id_deleted`). Nit: change the naming to `deleted_job_id`?

File: tests/stack/test_api.py [hunk 2]
```diff
@@ -152,9 +156,11 @@ def get_job(self, job_id, headers=None):
             raise HTTPError('Job not Found')
         else:
             return self.jobs_in_databricks[job_id]
-
+          
     def reset_job(self, data, headers=None):
-        if data[api.JOBS_RESOURCE_JOB_ID] not in self.jobs_in_databricks:
+        if data[api.JOBS_RESOURCE_JOB_ID] == job_id_deleted:
+            raise HTTPError
```
Comment (potcharapol-databricks) [inline tests/stack/test_api.py]: Nit: Add a message

File: databricks_cli/stack/api.py [hunk 3]
```diff
@@ -294,10 +295,14 @@ def _update_job(self, job_settings, job_id, headers=None):
...
-
-        self.jobs_client.reset_job({'job_id': job_id, 'new_settings': job_settings},
+        try:
+            self.jobs_client.reset_job({'job_id': job_id, 'new_settings': job_settings},
                                    headers=headers)
-
+        except HTTPError:
+            raise StackError('Job ID could not be found in the workspace while the status file '
+                             'still contains the Job ID {}. Please resolve the inconsistency '
+                             'before proceeding. Aborting stack deployment ...'.format(job_id))
```
Comment (potcharapol-databricks) [inline databricks_cli/stack/api.py]: Nit: Maybe add some suggestions here. For example, `Please remove or modify the local status file...` would be great.

Comment (catherineta) [reply databricks_cli/stack/api.py]: Modified

File: databricks_cli/stack/api.py [hunk 4]
```diff
@@ -294,10 +295,15 @@ def _update_job(self, job_settings, job_id, headers=None):
         :param job_settings: job settings to update the job with.
         :param job_id: physical job_id of job in databricks server.
         """
-
-        self.jobs_client.reset_job({'job_id': job_id, 'new_settings': job_settings},
+        try:
+            self.jobs_client.reset_job({'job_id': job_id, 'new_settings': job_settings},
                                    headers=headers)
```
Comment (potcharapol-databricks) [inline databricks_cli/stack/api.py]: I think you need to indent this line too.

File: databricks_cli/stack/api.py [hunk 5]
```diff
@@ -294,10 +295,15 @@ def _update_job(self, job_settings, job_id, headers=None):
...
-
+        except HTTPError:
+            raise StackError('Job ID could not be found in the workspace while the status file '
+                             'still contains the Job ID {}. Please remove or make necessary '
+                             'changes to the current status file to resolve this inconsistency '
+                             'before proceeding. Aborting stack deployment ...'.format(job_id))
+
```
Comment (potcharapol-databricks) [inline databricks_cli/stack/api.py]: Blank line should not contain whitespaces.

File: tests/stack/test_api.py [hunk 6]
```diff
@@ -140,21 +143,25 @@
...
+        self.deleted_job_id = 111
 
     def get_job(self, job_id, headers=None):
         if job_id not in self.jobs_in_databricks:
             # Job created is not found.
             raise HTTPError('Job not Found')
         else:
             return self.jobs_in_databricks[job_id]
-
+
```
Comment (potcharapol-databricks) [inline tests/stack/test_api.py]: ditto

File: tests/stack/test_api.py [hunk 7]
```diff
@@ -140,21 +143,25 @@
                          ]
 }
 
-
```
Comment (potcharapol-databricks) [inline tests/stack/test_api.py]: Need two blank lines before class definition.

File: databricks_cli/stack/api.py (hunk 3 above)
Comment (xuanwang14) [reply databricks_cli/stack/api.py]: Replace "status file" -> "stack status"?
Minor: "Job ID {} in stack status could not be found in the workspace. Please ... "

File: tests/stack/test_api.py [hunk 8]
```diff
@@ -154,7 +158,11 @@ def get_job(self, job_id, headers=None):
             return self.jobs_in_databricks[job_id]
 
     def reset_job(self, data, headers=None):
-        if data[api.JOBS_RESOURCE_JOB_ID] not in self.jobs_in_databricks:
+        if data[api.JOBS_RESOURCE_JOB_ID] == self.deleted_job_id:
+            raise HTTPError('Job ID could not be found in the workspace while '
```
Comment (xuanwang14) [inline tests/stack/test_api.py]: why do you need this change here? The test job client should not be affected by changes in stack CLI right?

File: tests/stack/test_api.py [hunk 9]
```diff
@@ -235,6 +243,15 @@ def test_deploy_job(self, stack_api):
         with pytest.raises(StackError):
             stack_api._deploy_job(alt_test_job_settings)
 
+        # TEST CASE 5
+        # If a databricks_id is deleted from the workspace while the status file contains that
```
Comment (xuanwang14) [inline tests/stack/test_api.py]: This test is more about inconsistency caused by "job not found", not about "deleted job". I would phrase the test to
"If a databricks_id is not found in workspace, then abort"
and then use variables such as 
`nonexisting_databricks_id`

**Comment by xuanwang14** [comment]
Looks good!

**Pull request merged**